File: 3-CODING-BLOCK/1-HASHING/1-CONTAINS-DUP-1.js
const duplicate = (nums) => {
  let x = {};
  for (let i = 0; i < nums.length; i++) {
    if (x[nums[i]]) {
      return true;
    } else {
      x[nums[i]] = true;
    }
  }

  return false;
};

nums = [1, 2, 3, 4];
console.log(duplicate(nums));

File: 3-CODING-BLOCK/1-HASHING/CODE/1-CONTAINS-DUP-1.py
def containsDuplicate( nums) :
    x={}
    for i in range(len(nums)):
        if nums[i] not in x:
            x[nums[i]]=1
        else:
            return True
    print(x)
    return False
    
nums=[1,2,1,2]
print(containsDuplicate(nums))